components/automate-ui/src/app/pages/project/rules/project-rules.component.ts

```diff
@@ -171,36 +186,44 @@ export class ProjectRulesComponent implements OnInit, OnDestroy {
     return this.conditions;
   }
 
-  updateConditionValues(condition): void {
-    if (condition.controls.operator.value === 'MEMBER_OF') {
-      condition.controls.values.value =
-        (typeof condition.controls.values.value === 'string')
-          ? [condition.controls.values.value]
-          : condition.controls.values.value.split(',').map((v) => v.trim());
-    } else {
-      condition.controls.values.value =
-        (typeof condition.controls.values.value === 'string')
-          ? condition.controls.values.value
-          : condition.controls.values.value.join(' ');
-    }
-  }
-
-  getConditionValue(value): string {
-    return (typeof value === 'string') ? value : value.join(', ');
-  }
-
   createRule() {
-    this.store.dispatch(new CreateRule({project_id: this.project.id, rule: this.ruleForm.value}));
+    // TODO: Generate the id!
+    this.store.dispatch(
+      new CreateRule({
+        project_id: this.project.id,
+        rule: this.convertToRule()
+      }));
   }
 
   updateRule() {
-    const updatedRule = this.ruleForm.value;
-    updatedRule.id = this.rule.id;
-    updatedRule.project_id = this.rule.project_id;
+    const updatedRule = this.convertToRule();
     this.store.dispatch(new UpdateRule({ rule: updatedRule }));
     this.store.dispatch(new GetRulesForProject({ project_id: this.rule.project_id }));
   }
 
+  convertToRule(): Rule {
+    const ruleValue = this.ruleForm.value;
+    const conditions: Condition[] = [];
+    ruleValue.conditions.forEach(c => {
+      conditions.push(<Condition>{
+        attribute: c.attribute,
+        operator: c.operator,
+          // Convert values string to storage format
+        values: c.operator === <ConditionOperator>'EQUALS'
+          ? [c.values.trim()]
+          : c.values.split(/,\s*/).map(v => v.trim())
```

Comment:
... And this is the other direction: translating the displayed string value back to an appropriate array for storage.